test/box-luatest/gh_6857_tuple_ext_validation_test.lua

```diff
@@ -0,0 +1,60 @@
+local t = require('luatest')
+local cluster = require('test.luatest_helpers.cluster')
+local server = require('test.luatest_helpers.server')
+
+local g = t.group('gh-6857-tuple-ext-validation')
+
+local socket = require('socket')
+local uri = require('uri')
+
+g.before_all = function()
+    g.cluster = cluster:new({})
+    local box_cfg = {
+        log_level = 6,
+    }
+    g.server = g.cluster:build_and_add_server({
+        alias='default',
+        box_cfg = box_cfg,
+    })
+    g.cluster:start()
+    g.listen = uri.parse(server.build_instance_uri('default'))
+    g.server:exec(function()
+        box.schema.space.create('test')
+        box.space.test:create_index('pk',{parts={{1, 'decimal'}}})
+        box.space.test:insert{require('decimal').new(1)}
+    end)
+end
+
+g.after_all = function()
+    g.cluster:drop()
+end
+
+-- Check extension types in tuple handling: previously mp_check() simply
+-- iterated over tuple field headers without looking into the field contents.
+-- This was ok for plain types, like MP_STR, MP_INT and others, because any
+-- msgpack encoded string of any of these types represents a valid value.
+-- This isn't true for extension types, however. Types like decimal, uuid and
+-- others have complex internal structure, which requires additional validation.
+g.test_extension_tuple_validation = function(g)
+    local sock = socket.tcp_connect(g.listen.host, g.listen.service)
```

Comment:
We have a special 'inject' method in the net.box module, which is used in tests for writing raw binary data to a connection. Maybe, better use it instead of a raw socket. It might be worthwhile to add a wrapper for the `_request(M_INJECT)` to the net.box module while we are at it, because using the `_request` remote method directly isn't particularly convenient.

Reply:
Done. Please, see 2 separate commits regarding net.box